animations/concepts.md

```diff
@@ -0,0 +1,298 @@
+---
+layout: page
+title: "Flutter animation concepts"
```

Comment:
I wish there were diagrams in the concepts doc to help illustrate how different concepts are related to one another, a pain-point both study participants and internal users reported. Without visual explanations, this doc doesn't look very different from clicking into each class's API doc and read the descriptions there. I made an attempt at this a while ago, please take a look: https://github.com/flutter/flutter/issues/13931#issuecomment-367849618 I'd be happy to work with you on adding more visual explanations.